helper/restore_helper.go

```diff
@@ -326,6 +365,15 @@ func doRestoreAgent() error {
 
 		logVerbose(fmt.Sprintf("Oid %d, Batch %d: End batch restore", tableOid, batchNum))
 
+		// On resize restore reader might be nil.
+		if !*singleDataFile && !(*isResizeRestore && contentToRestore >= *origSize) {
+			errPlugin := readers[contentToRestore].waitForPlugin()
+			readers[contentToRestore].logPlugin()
+			if errPlugin != nil {
+				err = errPlugin
```

Comment:
Why do you overwrite err coming from copyData or copyAllData?

Reply:
I've made this change in order to make the helper fail (and generate the error file) if plugin failed. I guess we could improve it by checking if `err` is `nil` and assign `err` only if it was `nil`....depending on which err we consider more important to report further. 

And actually it will override the err returned by `flushAndCloseRestoreWriter`. Which seems to be a separate problem - in case copyData or copyAllData failes, their err is now overwritten by flushAndCloseRestoreWriter, which I guess could successfully flush everything and return `nil` as error...making the helper to continue the loop as usual. What do you think about it?

Reply:
> Which seems to be a separate problem

Yes, https://github.com/arenadata/gpbackup/pull/100